File: backend/src/SurveyApp.Core/Entities/Question.cs
using SurveyApp.Core.Enums;

namespace SurveyApp.Core.Entities;

public class Question
{
    public Guid Id { get; set; }
    public string Text { get; set; } = string.Empty;
    public QuestionType Type { get; set; } = QuestionType.SingleChoice;

    public Guid? AnswerTemplateId { get; set; }
    public AnswerTemplate? AnswerTemplate { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public Guid? OwnerId { get; set; }
    public User? Owner { get; set; }
}

File: frontend/src/types/question.ts
export type QuestionType = 'SingleChoice' | 'MultipleChoice' | 'FreeText';

export interface Question {
    id: string;
    text: string;
    type: QuestionType;
    answerTemplateId: string | null;
    answerTemplateName: string | null;
    isDefault: boolean;
    isMine: boolean;
}

export interface CreateQuestionRequest {
    text: string;
    type: QuestionType;
    answerTemplateId: string | null;
}

export interface UpdateQuestionRequest {
    text: string;
    type: QuestionType;
    answerTemplateId: string | null;
}
